### PR TITLE
Added ModuleManager and TaskManager

### PR DESCRIPTION
**Pull request by SimJunYou**
Updated ModuleManager and TaskManager.

ModuleManager:
- getModule (takes moduleCode)
- edit (takes moduleCode and new Module object)
- add (takes new Module object)
- delete (takes moduleCode)
- contains (takes moduleCode)

TaskManager:
- getTask (takes taskId)
- edit (takes taskId and new Task object)
- add (takes new Task object)
- delete (takes taskId)

### CODE REVIEW

File: src/main/java/seedu/duke/data/ModuleManager.java [hunk 1]
```diff
@@ -4,12 +4,10 @@
 import seedu.duke.exception.DuplicateDataException;
 import seedu.duke.exception.ModuleNotProvidedException;
 
-import java.util.ArrayList;
 import java.util.HashMap;
 
 public class ModuleManager {
-    private static ArrayList<Module> moduleList;
-    private static HashMap<String, String> modulesMap;
+    private static HashMap<String, Module> modulesMap = new HashMap<>(); // Main module list. Maps module code to module object.
```
Comment (wangqinNick) [inline src/main/java/seedu/duke/data/ModuleManager.java:10]: I think it should contain two module list, (or maps) 
One module map is for storing the loaded module map from the NUSMODS
The other one is for storing the modules added by user.

Comment (SimJunYou) [reply src/main/java/seedu/duke/data/ModuleManager.java:10]: Okay, I will add that in a later PR

**Comment by Aseanseen** [comment]
Looks good to merge

**Review by Aseanseen — APPROVED** [review]
Looks good to go! Merge if checks pass

**Pull request merged**